packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Models/Types/ModelTypeProvider.cs

```diff
@@ -121,5 +121,203 @@ private bool PropertyHasSetter(CSharpType type, InputModelProperty prop)
+                }
+                else
+                {
+                    // For classes, only required + not readonly + not initialization value + not discriminator could get into the public ctor
+                    // For structs, all properties must be set in the public ctor
+                    if (IsStruct || (property is { IsRequired: true, IsDiscriminator: false } && initializationValue == null))
+                    {
+                        if (!property.IsReadOnly)
+                        {
+                            constructorParameters.Add(parameter with { Type = parameter.Type.InputType });
+                        }
+                    }
+                }
+            }
+
+            return constructorParameters;
+        }
+
+        private static ValidationType GetParameterValidation(InputModelProperty property, CSharpType propertyType)
+        {
+            // We do not validate a parameter when it is a value type (struct or int, etc)
+            if (propertyType.IsValueType)
+            {
+                return ValidationType.None;
+            }
+
+            // or it is readonly
+            if (property.IsReadOnly)
+            {
+                return ValidationType.None;
+            }
+
+            // or it is optional
+            if (!property.IsRequired)
+            {
+                return ValidationType.None;
+            }
+
+            // or it is nullable
+            if (propertyType.IsNullable)
+            {
+                return ValidationType.None;
+            }
+
+            return ValidationType.AssertNotNull;
+        }
+
+        private CSharpMethod BuildInitializationConstructor()
+        {
+            var accessibility = DeclarationModifiers.HasFlag(TypeSignatureModifiers.Abstract)
+                ? MethodSignatureModifiers.Protected
+                : _inputModel.Usage.HasFlag(InputModelTypeUsage.Input)
+                    ? MethodSignatureModifiers.Public
+                    : MethodSignatureModifiers.Internal;
+            var constructorParameters = BuildConstructorParameters(false);
+
+            var constructor = new CSharpMethod(
+                signature: new ConstructorSignature(
+                    Type,
+                    $"Initializes a new instance of {Type:C}",
+                    null,
+                    accessibility,
+                    constructorParameters),
+                body: new MethodBodyStatement[]
+                {
+                    new ParameterValidationBlock(constructorParameters),
+                    GetPropertyInitializers(constructorParameters)
+                },
+                kind: CSharpMethodKinds.Constructor);
+
+            return constructor;
+        }
+
+        private CSharpMethod? BuildSerializationConstructor()
+        {
+            // The property bag never needs deserialization, therefore we return the null here
+            if (_inputModel.IsPropertyBag)
```

Comment:
I think we do not even have a design for property bag right now.
I suggest we remove related properties completely for clarity.